frame/pallet-account/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Custom origin type.
+        type CustomOrigin: Parameter + TryInto<Self::AccountId> + MaxEncodedLen;
+        /// Parameter that defines different origin options and how to create it.
+        type CustomOriginKind: Parameter + OriginDeriving<Self::AccountId, Self::CustomOrigin>;
+        /// The runtime origin type.
+        type RuntimeOrigin: From<Self::CustomOrigin>
+            + From<frame_system::RawOrigin<Self::AccountId>>;
+        /// The overarching call type.
+        type RuntimeCall: Parameter
+            + Dispatchable<RuntimeOrigin = <Self as Config>::RuntimeOrigin>
+            + GetDispatchInfo
+            + From<frame_system::Call<Self>>
+            + IsSubType<Call<Self>>
+            + IsType<<Self as frame_system::Config>::RuntimeCall>;
+        /// General event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Origin with given index not registered.
+        UnregisteredOrigin,
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(crate) fn deposit_event)]
+    pub enum Event<T: Config> {
+        NewOrigin {
+            account: T::AccountId,
+            origin: T::CustomOrigin,
+        },
+        ProxyCall {
+            origin: T::CustomOrigin,
+            result: DispatchResult,
+        },
+    }
+
+    #[pallet::origin]
+    pub type Origin<T> = <T as Config>::CustomOrigin;
+
+    /// Account origins
+    #[pallet::storage]
+    pub type AccountOrigin<T: Config> =
+        StorageDoubleMap<_, Blake2_128Concat, T::AccountId, Twox64Concat, u32, T::CustomOrigin>;
+
+    /// Account last origin index
+    #[pallet::storage]
+    pub type AccountLastOrigin<T: Config> =
+        StorageMap<_, Twox64Concat, T::AccountId, u32, ValueQuery>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Derive new origin for account.
+        ///
+        /// The dispatch origin for this call must be _Signed_.
+        #[pallet::weight(
+            T::WeightInfo::new_origin()
+                .saturating_add(T::DbWeight::get().reads_writes(1, 2))
+        )]
+        #[pallet::call_index(0)]
+        pub fn new_origin(
+            origin: OriginFor<T>,
+            origin_kind: T::CustomOriginKind,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            let next_index = AccountLastOrigin::<T>::get(&who);
+            let new_origin = origin_kind.derive(&who, next_index);
+            AccountOrigin::<T>::insert(&who, next_index, new_origin.clone());
+            AccountLastOrigin::<T>::insert(&who, next_index + 1);
```

Comment:
This is still free - so my comment from 2 months ago is still valid.
You should charge  _rent fee_ for these two items since you're inflating the database.

Check how `pallet-assets` is doing it, or any other pallet in `substrate` that creates some storage items (e.g. `pallet-identity`, `pallet-multisig`, etc.).

Reply:
Technically, you pay transaction fee to cover it (prevent spam), it’s not required at all, just one of ways to do it. I’m not sure it’s required for the prototype.

Reply:
My personal opinion - better charge fee than just lock token forever. Charged fee much more controllable, it could be burn, redistributed or whatever you want. But locked deposit is just locked.

Reply:
The way it's implemented now - sure, that's viable option. It still needs to be added though since it's missing now.

Reply:
Not sure something is missing now, what precisely you are talking about?

Reply:
I'm talking about payment for the created storage.

> Technically, you pay transaction fee to cover it (prevent spam), it’s not required at all, just one of ways to do it. I’m not sure it’s required for the prototype.

Given that our handling of fees is copy/paste of Polkadot, it is how we do things, isn't it?
Don't understand why this pallet should be any different.
Are you proposing we mix it up, some pallets charging for rent fee, while some don't? 🤷‍♂️ 

It's been on review for months, and better to do it right immediately then have someone else fix it later, add storage migration logic, etc. Doesn't matter if it's prototype or not, IMO.